### PR TITLE
 Fix ArrayIndexOutOfBoundsException with compact SVG arc notation

### LINKED ISSUE #1244: Loading SVG fails for some files
### Most appropriate sub-area of Processing 4?

IO ?

### Processing version

4.4.7

### Operating system

Windows/macOS/Linux

### Bug description

Unable to load some SVG files, such as https://commons.wikimedia.org/wiki/File:Aiga_departingflights.svg  you get something similar to:

`ArrayIndexOutOfBoundsException: Index 43 out of bounds for length 43`

### Steps to reproduce this

1. Launch PDE, create a sketch, download https://commons.wikimedia.org/wiki/File:Aiga_departingflights.svg  and add it to the `data` folder of your sketch.

2. Run the snippet below:

```processing
void setup(){
  size(500, 500);
  PShape svg = loadShape("Aiga_departingflights.svg");
  shape(svg);  // will crash before this line... 
}
```

### Additional context

This looks a bit like https://github.com/processing/processing4/issues/750 but the "SVG path notation parsing issue" that causes it to fail seems different.

The provided failing SVG looks like this:

Image

```svg
<svg xmlns="http://www.w3.org/2000/svg" version="1.0" viewBox="0 0 29 29">
<path d="m0 6 3-2 15 4 7-7a2 2 0 013 3l-7 7 4 15-2 3-7-13-5 5v4l-2 2-2-5-5-2 2-2h4l5-5z"/>
</svg>
```
If you scale it a bit on Inkscape you get a different path notation and it works fine:

```svg
<svg xmlns="http://www.w3.org/2000/svg" version="1.0" viewBox="0 0 29 29">
<path d="m 0,5.9994379 2.9997,-1.9998 14.9985,3.9996 6.9993,-6.99930004 a 2.1211082,2.1211082 0 0 1 2.9997,2.99970004 l -6.9993,6.9993001 3.9996,14.9985 -1.9998,2.9997 -6.9993,-12.9987 -4.9995,4.9995 v 3.9996 l -1.9998,1.9998 -1.9998,-4.9995 -4.9995,-1.9998 1.9998,-1.9998 h 3.9996 l 4.9995,-4.9995 z"/>
</svg>
```

### Would you like to work on the issue?

No, I’m just reporting the issue

### PR DESCRIPTION
**Pull request by hxrshxz**
Fixes #1244

## Summary

Adds support for compact SVG arc notation where flags and coordinates are concatenated (e.g., `a2 2 0 013 3` instead of `a 2 2 0 0 1 3 3`). Previously crashed with `ArrayIndexOutOfBoundsException`.

## Changes

- **`core/src/processing/core/PShapeSVG.java`**: Added `isCompactArcNotation()` helper method to detect and parse concatenated arc flags/coordinates for both `A` and `a` commands
- **`core/test/processing/core/PShapeSVGPathTest.java`**: Added 4 test cases covering compact notation, standard notation, and edge cases

## Before

https://github.com/user-attachments/assets/d16fd9f4-f660-4332-81b3-4874a5266df3

## After

https://github.com/user-attachments/assets/0f5de7c3-d813-4b29-895b-c75e5909efc5

### CODE REVIEW

**Comment by hxrshxz** [comment]
@catilac have a look :)

**Comment by Stefterv** [comment]
Hi @hxrshxz great work! Thank you! 

Looking at the tests, the one thing I would say to be missing is a step that actually checks if the parsing occurred correctly by looking at the numbers that the parser spits out, I would say just checking if the shape parses is not enough.

**Comment by hxrshxz** [comment]
Thanks @Stefterv I have added tests to verify the actual parsed endpoint values, not just that the shape doesn't crash

File: core/src/processing/core/PShapeSVG.java [hunk 1]
```diff
@@ -961,14 +961,36 @@ else if (lexState == LexState.EXP_HEAD) {
...
+          } else {
+            endX = PApplet.parseFloat(pathTokens[i + 5]);
+            endY = PApplet.parseFloat(pathTokens[i + 6]);
+            tokenOffset = -1;
+          }
+        } else {
+          fa = PApplet.parseFloat(token4) != 0;
+          fs = PApplet.parseFloat(pathTokens[i + 5]) != 0;
+          endX = PApplet.parseFloat(pathTokens[i + 6]);
+          endY = PApplet.parseFloat(pathTokens[i + 7]);
```
Comment (catilac) [inline core/src/processing/core/PShapeSVG.java]: what is the significance of the 5, 6 and 7 index offsets for pathTokens?S Specifically `pathTokens[i+5]` etc

File: core/src/processing/core/PShapeSVG.java [hunk 2]
```diff
@@ -1054,6 +1096,29 @@ private void parsePathMoveto(float px, float py) {
...
+    }
+    return token.length() > 1 &&
+           (token.charAt(0) == '0' || token.charAt(0) == '1') &&
+           (token.charAt(1) == '0' || token.charAt(1) == '1') &&
+           (token.length() == 2 ||
+            (token.length() > 2 && (
+              Character.isDigit(token.charAt(2)) ||
+              token.charAt(2) == '+' ||
+              token.charAt(2) == '-' ||
+              token.charAt(2) == '.')));
```
Comment (catilac) [inline core/src/processing/core/PShapeSVG.java:1140]: this makes sense but could use some comments to describe what is going on

**Review by catilac — COMMENTED** [review]
Looks good! I have some questions. Sorry to take so long with this.

File: core/src/processing/core/PShapeSVG.java (hunk 1 above)
Comment (hxrshxz) [reply core/src/processing/core/PShapeSVG.java]: The i + 5, i + 6, and i + 7 offsets correspond to the standard SVG arc command structure which takes 7 arguments: (rx ry x-axis-rotation large-arc-flag sweep-flag x y).

When parsing the A (or a) command at index i:

i + 1: rx
i + 2: ry
i + 3: x-axis-rotation (angle)
i + 4: large-arc-flag (token4)
Therefore, the remaining arguments are at:

i + 5: sweep-flag (fs)
i + 6: x (endX)
i + 7: y (endY)

**Comment by hxrshxz** [comment]
really sorry for taking so long

File: core/src/processing/core/PShapeSVG.java (hunk 2 above)
Comment (hxrshxz) [reply core/src/processing/core/PShapeSVG.java:1140]: done

**Review by catilac — APPROVED** [review]

**Comment by catilac** [comment]
@hxrshxz this is approved! yay. we are going to do some additional hand testing before it gets merged in. we have an upcoming 4.5 release so i'm being extra careful. but thank you so much! sorry to take so long in accepting this.
